diff --git a/backend/postgres/postgres.go b/backend/postgres/postgres.go
--- a/backend/postgres/postgres.go
+++ b/backend/postgres/postgres.go
@@ -1081,8 +1081,10 @@ func (be *postgresBackend) getActivityWorkItem(ctx context.Context) (*backend.Act
 		`UPDATE NewTasks SET LockedBy = $1, LockExpiration = $2, DequeueCount = DequeueCount + 1
 		WHERE SequenceNumber = (
 			SELECT SequenceNumber FROM NewTasks T
 			WHERE T.LockExpiration IS NULL OR T.LockExpiration < $3
+			ORDER BY T.SequenceNumber ASC
+			FOR UPDATE SKIP LOCKED
 			LIMIT 1
 		) RETURNING SequenceNumber, InstanceID, EventPayload`,
 		be.workerName,
